File: Graduating-Server/src/main/kotlin/com/bestswlkh0310/graduating/graduatingserver/entity/SchoolType.kt
package com.bestswlkh0310.graduating.graduatingserver.entity

enum class SchoolType {
    HIGH,
    MIDDLE,
    ELEMENTARY;

    companion object {
        fun of(string: String) = entries.first {
            it.name == string
        }

        fun ofKorean(string: String) = entries.first {
            it.korean() == string
        }
    }

    fun korean() = when(this) {
        HIGH -> "고등학교"
        MIDDLE -> "중학교"
        ELEMENTARY -> "초등학교"
    }
}

File: Graduating-iOS/Model/School/SchoolType.swift
//
//  SchoolType.swift
//  Model
//
//  Created by hhhello0507 on 8/28/24.
//

import Foundation

public enum SchoolType: String, ModelProtocol {
    case HIGH
    case MIDDLE
    case ELEMENTARY
    
    public var limit: Int {
        switch self {
        case .HIGH: 3
        case .MIDDLE: 3
        case .ELEMENTARY: 6
        }
    }
}
